### DISCUSSION TITLE
What's relation of rust-libp2p and the libp2p in this project

### DISCUSSION THREAD

**Discussion by tinymahua** [n0-computer/iroh#1277] [Q&A] [answered]
I had researched a few time at [rust-libp2p], then I recently found this project.
I am very interesting this iroh project, I want to kown what relation of  the libp2p implement in iroh and rust-libp2p.
Can I get these information by someway?

**Comment by dignifiedquire** [comment, marked as answer]
I have previously worked with and on libp2p and rust-libp2p. Iroh is a different take on some similar issues you run into when trying to build more peer to peer based applications. But iroh itself currently does not use libp2p, and tries to solve for a specific set of protocols used on the wire. libp2p gives you a set of different protocols and extensions to connect peers in a peer to peer fashion. 

Historically the first version of iroh (now known as beetle), which aimed at being a fully compatible IPFS node did use rust-libp2p. But as said above, the current version does not anymore.

**Reply by tinymahua** [reply]
Thank you.
